### ISSUE TITLE
Faster initial setup by avoiding type check during build

### ISSUE THREAD

**Issue by 5ebastianMeier** [process-engine/process_engine_meta#31]
#### Describe your issue

I noticed an option on the TypeScript compiler that allows to compile each module individually. This results in a massive performance boost, since only the transpilation is done - not the type check.
The option is called `--isolatedModules` (https://www.typescriptlang.org/docs/handbook/compiler-options.html)

Since its the job of Jenkins to ensure the sources build correctly and the tests are running, I think we could improve the initial setup time for the meta repo tremendously by allowing this option using a special flag.

In my initial tests the results where ~5s before / ~300ms after **per** repository.
So we could save about 5 to 10 minutes on the initial setup (initial setup includes cloning, installing and building everything).

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

**Comment by cmg-dev** [comment]
@5ebastianMeier @heikomat What is the status in this one?

**Comment by 5ebastianMeier** [comment]
This is supported now by using this environment variable in conjunction with `gulptraum-typescript@1.5.0`:

```
GULPTRAUM_TRANSPILE_ONLY=1 gulp build
```

**Comment by ElRaptorus** [comment]
I'd advise caution on using this.

During the last major release cycle, I noticed a rather strange side effect:

When having a large setup (i.e. all the packages from `@essential-projects` and/or `@process-engine`), then adding this flag to the build-process will cause some of the `.d.ts` files not to be created. 
This in turn will break the build for all modules which depend on the affected modules.

I haven't been able to reproduce this with any real certainty.
However, I did notice that the problem seems to start when building the 15th-16th module of your setup. 
I can't say if this effect is limited to these modules, because building the subsequent modules all failed anyway, because the dependency chain was broken.
